src/Chapter2.hs

```diff
@@ -841,7 +861,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n ls
+  | n < 0 = []
+  | otherwise = subList n (n + length ls - 1) (cycle ls)
```

Comment:
Nice reusage of `subList` 🧠 

In order to make this solution more efficient on case edge-cases, you can return the list itself, when `n == 0` 🙂 
But your implementation already covers all cases! This is just an idea for performance 🚤